### ISSUE TITLE
[Feature]: Add support for attention score output

### ISSUE THREAD

**Issue by WoutDeRijck** [vllm-project/vllm#11365]
## Problem
vLLM currently doesn't provide access to attention scores during inference, which are essential for model analysis and interpretability research. #11862 

## Feature Request
Add the ability to retrieve attention scores during model inference, similar to HuggingFace's output_attentions=True parameter.

## Motivation
Need to analyze token-level relationships in model outputs
Required for building visualization tools and debugging model behavior
Critical for research into attention mechanisms

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by Dineshkumar-Anandan-ZS0367** [comment]
You are asking output_attentions=True or return_cross_attentions=True for getting coordinates right.

These only given by vision encoder decoder models or cross encoder models.

Which model?

**Comment by Dineshkumar-Anandan-ZS0367** [comment]
Apologise by mistakes. I have integrated score using tensor logits already. Thanks!

**Comment by WoutDeRijck** [comment]
I do not need the logits as well. I need the attention scores.

**Comment by HuiSiqi** [comment]
Any update of this? I also need to visualize the attention scores of decoder-based models.

**Comment by oneonlee** [comment]
Any update on this?

**Comment by mariamavagyan** [comment]
Any update on this? It would be great to this feature!

**Comment by faranten** [comment]
Any update on this? It is really a key feature for researchers!

**Comment by aciokler** [comment]
Any updates? This feature would be extremely useful!

**Comment by bruce2233** [comment]
Any update on this? It would be great to this feature!

**Comment by MTDickens** [comment]
Any update on this? I think many researcher (including me ofc) really do need this feature.

**Comment by MTDickens** [comment]
Again any updates

**Comment by SophieOstmeier** [comment]
Any update on this? It would be great to this feature! Thank you very much for considering!
